### ISSUE TITLE
Add option to blacklist words

### ISSUE THREAD

**Issue by zydd** [birtles/rikaichamp#124]
E.g.: press X while the popup is open to not show the popup for that word again.
It would also be interesting to be possible to import/export a list of known words.

I leave rikaichamp on most of the time and it's a bit disruptive to have the popup show when clicking a link or accidentally hovering words I already know.

**Comment by zydd** [comment]
I could try coding this in the next few weeks but I can't promise the code will be great since I'm not used to programming in JS/TypeScript.

**Comment by birtles** [comment]
Thanks, that's an interesting suggestion. Out of interest, have you tried the <kbd>Ctrl</kbd> "Hold to show popup" setting? That's what I use. I have Rikaichamp always enabled and then just hold <kbd>Ctrl</kbd> down while hovering over a word or two I don't know.

If you decide to try writing a patch, it might be good to wait a few days. I have a very large change underway on the `ng-kanji` branch which I should merge in the next few days that might conflict with any changes you make.

Also, bear in mind that part of the difficulty of such a patch might be reliably synchronizing the list of known words. I'm not sure if there are limits to the size of sync storage values.

**Comment by zydd** [comment]
> have you tried the Ctrl "Hold to show popup" setting?

Thanks for the suggestion! I'll be using this option from now on.

> Also, bear in mind that part of the difficulty of such a patch might be reliably synchronizing the list of known words. I'm not sure if there are limits to the size of sync storage values.

According to [this link](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync) the storage limit is 100KB, that would be enough to save a reasonably large number of entries even without compression.

Furthermore, it seems `storage.sync` only has `get()` and `set()` methods, the _syncing_ itself is done by the browser.

**Comment by birtles** [comment]
> Furthermore, it seems `storage.sync` only has `get()` and `set()` methods, the _syncing_ itself is done by the browser.

Yeah, I was more thinking about if you had conflicting additions/deletions to the list but I guess it would be safe to ignore that situation and just go with a last-one-wins strategy.
